code/game/objects/items/grenade_box.dm

```diff
@@ -0,0 +1,82 @@
+///OBJECT DEFINITION///
+/obj/item/grenade_box
+	name = "Grenade Box"
+	desc = "A box used to conveniently and (relatively) safely transport grenades."
+	icon = 'icons/obj/warfare.dmi'
+	icon_state = "grenade_box"
+	var/list/grenades = list()
+	var/open = FALSE
+
+///INTERACTIONS///
+
+/obj/item/grenade_box/attackby(obj/item/G as obj, mob/user as mob) //PUTTING GRENADES IN
+	if(open == FALSE)
+		to_chat(user,SPAN_WARNING("I can't use it when it's closed!"))
+
+	else if(istype(G,/obj/item/grenade))
+		insert_grenade(G,user)
+
+/obj/item/grenade_box/RightClick(mob/user) //Opening and closing
```

Comment:
You need a "can interact" check here. Right click will always function at any distance otherwise. Especially since you are not calling the parent.